app_server/app/services/UserService.js

```diff
@@ -84,4 +85,67 @@ Class.getNumberOfUsers = function() {
   return Storage.getNumberOfUsers();
 };
 
+///// VIEW RELATED ////
+Class.createView = function(userId, streamId) {
+  return Storage.createView(userId, streamId).then(function(res) {
+    if (!res) {
+      logger.error('Unable to create view');
+
+      return Promise.resolve(new CustomError
+        .NotFoundError('Stream not found'));
+    }
+    return res.dataValues;
+
+  }).catch(function(err) {
+    logger.error('Unable to create view: %j', err);
+
+    if (err.name === 'TypeError') {
+      return Promise.resolve(new CustomError
+        .NotFoundError('User not found'));
+    } else {
+      Promise.resolve(new CustomError.UnknownError());
+    }
+  });
+};
+
+/**
+ * Gets the list of users watching a particular stream.
+ * @param streamId {string}
+ * @return {Promise<Array<User>>||null}
+ */
+Class.getListOfUsersViewingStream = function(streamId) {
+  logger.debug('Getting list of users watching stream: %s', streamId);
+
+  return Storage.getListOfUsersViewingStream(streamId)
+    .then(function receiveResult(result) {
+      if (result) {
+        return result.map(
+          function(singleUser) {
+            singleUser = singleUser.dataValues;
+            delete singleUser.View;
+            return Utility.formatUserObject(singleUser);
+          });
+      } else {
+        return null;
+      }
```

Comment:
No need the `else` here i.e. this structure is more preferred when the `if` block already terminates using a `return` statement since it's less nested:
```javascript
if (...) {
  return ...;
}
return ...;
```